lib/HM/Harmony_Search.hpp

```diff
@@ -0,0 +1,206 @@
+
+#ifndef LIB_HARMONY_SEARCH_HPP_
+#define LIB_HARMONY_SEARCH_HPP_
+
+#define _USE_MATH_DEFINES
+
+#include "./filter_param.hpp"
+#include "./opt_helper.hpp"
+
+#include <stdlib.h>
+#include <math.h>
+#include <stdio.h>
+#include <iostream>
+#include <fstream>
+#include <complex>
+#include <vector>
+#include <time.h>
+#include <random>
+struct Harmony_Search_Parameter
+{
+    protected:
+
+    //ハーモニーサーチのパラメータ
+    int Time_Max;   //更新回数
+    int Harmony_Size; //ハーモニーメモリのサイズ
+    double Bandwidth; //バンド幅
+    double R_a;       //選択比率(大きければ大きい程、ハーモニーメモリ内が使われる)
+    double R_p;       //値調整比率(大きければ大きい程、値を調整する確率が上がる)
```

Comment:
```suggestion
    //ハーモニーサーチのパラメータ
    size_t max_iter;   //更新回数
    size_t harmony_size; //ハーモニーメモリのサイズ
    double band_width; //バンド幅
    double R_a;       //選択比率(大きければ大きい程、ハーモニーメモリ内が使われる)
    double R_p;       //値調整比率(大きければ大きい程、値を調整する確率が上がる)
```

R_a, R_pはなんか名前変えたいよね
単文字は嫌だ